library/core/src/cell/once.rs

```diff
@@ -298,3 +298,7 @@ impl<T> const From<T> for OnceCell<T> {
         OnceCell { inner: UnsafeCell::new(Some(value)) }
     }
 }
+
+// Just like for `Cell<T>` this isn't needed, but results in nicer error messages.
+#[unstable(feature = "once_cell", issue = "74465")]
+impl<T> !Sync for OnceCell<T> {}
```

Comment:
Actually... Removing this would be a breaking change: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=1f4ad2c6d839ce8a21237c4faa7b2e9e (tl;dr: you'll be able to impl trait for `<T: Sync>` and `OnceCell`).

Should we consult t-libs?

Reply:
Right, this is easy to prove: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=1ff2fdaa8ed22c14e94aaddb348ab850.